src/v/security/audit/schemas/hashing_utils.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "security/audit/schemas/types.h"
+#include "security/authorizer.h"
+
+namespace security::audit {
```

Comment:
The implementation might be a lot simpler with some helpers for hashing, perhaps something like:
```c++
namespace impl {
template<typename T>
concept is_absl_hashable = requires(T const& t) {
    { absl::Hash<T>{}(t) } -> std::convertible_to<std::size_t>;
};

template<typename T>
concept is_std_hashable = requires(T const& t) {
    { std::hash<T>{}(t) } -> std::convertible_to<std::size_t>;
};

template<typename T>
concept is_boost_hashable = requires(T const& t) {
    { boost::hash<T>{}(t) } -> std::convertible_to<std::size_t>;
};

template<typename T>
size_t hash_combine(size_t& seed, T const& t) {
    if constexpr (is_std_hashable<T>) {
        boost::hash_combine(seed, std::hash<T>{}(t));
    } else if constexpr (is_absl_hashable<T>) {
        boost::hash_combine(seed, absl::Hash<T>{}(t));
    } else if constexpr (is_boost_hashable<T>) {
        boost::hash_combine(seed, boost::hash<T>{}(t));
    } else {
        static_assert(always_false_v<T>, "No hasher found for T");
    }
    return seed;
}

} // namespace impl

template<typename... Args>
size_t hash_combine(size_t& seed, Args&&... args) {
    return (impl::hash_combine(seed, std::forward<Args>(args)), ...);
}
```
This would allow something like:
```diff
-    boost::hash_combine(h, std::hash<category_uid>()(ct_uid));
-    boost::hash_combine(h, std::hash<class_uid>()(c_uid));
-    boost::hash_combine(h, std::hash<severity_id>()(s_id));
+    hash_combine(h, ct_uid, c_uid, s_id);
```
or:
```diff
-        if (result.acl) {
-            boost::hash_combine(
-              h, absl::Hash<security::acl_entry>{}(*result.acl));
-        }
-        if (result.resource_pattern) {
-            boost::hash_combine(
-              h,
-              absl::Hash<security::resource_pattern>{}(
-                *result.resource_pattern));
-        }
+        hash_combine(h, result.acl, result.resource_pattern);
```

Reply:
Nice idea 👍